.ci/pipelines/e2e-tests-testdeleteservices.Jenkinsfile

```diff
@@ -0,0 +1,113 @@
+// This library overrides the default checkout behavior to enable sleep+retries if there are errors
+// Added to help overcome some recurring github connection issues
+@Library('apm@current') _
+
+def failedTests = []
+def lib
+
+pipeline {
+
+    agent {
+        label 'linux'
+    }
+
+    options {
+        timeout(time: 300, unit: 'MINUTES')
+    }
+
+    environment {
+        VAULT_ADDR = credentials('vault-addr')
+        VAULT_ROLE_ID = credentials('vault-role-id')
+        VAULT_SECRET_ID = credentials('vault-secret-id')
+        GCLOUD_PROJECT = credentials('k8s-operators-gcloud-project')
+    }
+
+    stages {
+        stage('Checkout from GitHub') {
+            steps {
+                checkout scm
+            }
+        }
+        stage('Load common scripts') {
+            steps {
+                script {
+                    lib = load ".ci/common/tests.groovy"
+                }
+            }
+        }
+       /* stage('Run Checks') {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            steps {
+                sh 'make -C .ci TARGET=ci-check ci'
+            }
+        }*/
+        stage("E2E tests") {
+            when {
+                expression {
+                    notOnlyDocs()
+                }
+            }
+            steps {
+                sh '.ci/setenvconfig e2e/master'
+                
+                sh 'echo TESTS_MATCH = TestDeleteServices >> .env'
+                sh 'echo E2E_SKIP_CLEANUP = true >> .env'
+                sh 'sed "s/CLUSTER_NAME=eck-e2e/CLUSTER_NAME=eck-debug-endpoints-e2e/" -i .env'
+                sh 'sed "s/clusterName: eck-e2e/clusterName: eck-debug-endpoints-e2e/" -i deployer-config.yml'
+                
+                script {
+                    env.SHELL_EXIT_CODE = sh(returnStatus: true, script: 'make -C .ci get-test-artifacts TARGET=ci-build-operator-e2e-run ci')
+
+                    //sh 'make -C .ci TARGET=e2e-generate-xml ci'
+                    //junit "e2e-tests.xml"
```

Comment:
Let's try to reactivate this?
Without generating the report, `failedTests` will still be empty and the k8s cluster will still be deleted.

We commented on it as it seemed to be the reason we got spammed but we didn't understand what was going on.
I think a compromise could be to move it to the `if (env.SHELL_EXIT_CODE! = 0) {`.

Reply:
Like that?

```
                    if (env.SHELL_EXIT_CODE != 0) {
                        sh 'make -C .ci TARGET=e2e-generate-xml ci'
                        junit "e2e-tests.xml"
                        failedTests = lib.getListOfFailedTests()
                    }
```